chain-connect/src/GalachainConnectTONClient.ts

```diff
@@ -0,0 +1,184 @@
+  #chainCodeUrl: string;
+  #accessToken: string | null = null;
+
+  get galachainAddress() {
+    return this.#tonAddress.replace("0x", "ton|");
+  }
+
+  get tonAddress() {
+    return this.#tonAddress;
+  }
+
+  get accessToken() {
+    return this.#accessToken;
+  }
+
+  constructor(chainCodeUrl: string, manifestUrl: string) {
+    // super();
+    const connector = new TonConnect({ manifestUrl });
+
+    this.#chainCodeUrl = chainCodeUrl;
+    this.initializeTONListeners(connector);
+  }
+
+  private initializeTONListeners(connector): void {
+    connector.onStatusChange((wallet) => {
+      if (!wallet) {
+        this.#accessToken = null;
+        return;
+      }
+
+      const tonProof = wallet.connectItems?.tonProof;
+
+      if (tonProof) {
+        if ("proof" in tonProof) {
+          this.#tonAddress = wallet.account.address;
+          this.#tonPublicKey = wallet.account.publicKey;
+          this.checkProof(tonProof.proof, wallet.account);
+          return;
+        }
+
+        console.error(tonProof.error);
+      }
+
+      if (!this.#accessToken) {
+        connector.disconnect();
+      }
+    });
+  }
+
+  private async checkProof(proof: TonProofItemReplySuccess["proof"], account: Account) {
+    // send proof to your backend
+    // e.g. myBackendCheckProof(tonProof.proof, wallet.account);
+    try {
+      const reqBody = {
+        address: account.address,
+        network: account.chain,
+        proof: {
+          ...proof,
+          state_init: account.walletStateInit
+        }
+      };
+
+      // Build function based on proof service at:
+      // https://github.com/ton-connect/demo-dapp-with-react-ui/blob/master/src/server/services/ton-proof-service.ts
+      const response = await (
+        await fetch(`${this.#chainCodeUrl}/ton-proof/checkProof`, {
+          method: "POST",
+          body: JSON.stringify(reqBody)
+        })
+      ).json();
+
+      if (response?.token) {
+        this.#accessToken = response.token;
+      }
+    } catch (e) {
+      console.log("checkProof error:", e);
+    }
+  }
+
+  public async send<T, U extends ConstructorArgs<ChainCallDTO>>({
+    url = this.#chainCodeUrl,
+    method,
+    payload,
+    sign = false,
+    headers = {}
+  }: {
+    url?: string;
+    method: string;
+    payload: U;
+    sign?: boolean;
+    headers?: object;
+  }): Promise<T> {
+    if (!this.#tonAddress) {
+      throw new Error("No account connected");
+    }
+
+    try {
+      if (sign === true) {
+        const dto = signatures.getPayloadToSign(payload);
+        // const dtoBuff = Buffer.from(dto)
+        // const messageCell = beginCell()
+        // .storeBuffer(dtoBuff)
+        // .endCell();
+
+        // const signer = await this.#provider.getSigner();
+        // const signature = await signer.provider.send("personal_sign", [this.#tonAddress, dto]);
+
+        // const signature =
+
+        // return await this.submit(url, method, { ...prefixedPayload, signature }, headers);
```

Comment:
yes, and here is the place where we need to get signature, and add it to DTO. Also we'll need to have an additional field in dto to determine the signing scheme.

This is the `enum` I'd like you to add to `dtos.ts`:

```typescript
export enum SigningScheme {
  ETH = "ETH",
  TON = "TON"
}
```

And this is the field to add to `ChainCallDTO` right after the `signerPublicKey`:

```typescript
  @JSONSchema({
    description:
      "Signing scheme used for the signature. ETH for Ethereum, TON for The Open Network." +
      `Default: ${SigningScheme.ETH}`
  })
  @IsOptional()
  @StringEnumProperty(SigningScheme)
  public signing?: SigningScheme;
```

I'm doing the same changes in #325, but I think you need it as well to implement the proper code without waiting till #325 is finished.

Reply:
Also one more thing for signing - I don't know how it looks from TON side, but signature verification may also have a seed/prefix, like `ton-connect` string here: https://github.com/ton-connect/demo-dapp-with-react-ui/blob/master/src/server/services/ton-proof-service.ts#L103

please add this (if needed) as a `prefix` in dto